injector/cpu_pressure.go

```diff
@@ -7,155 +7,93 @@ package injector
+		config:                   config,
+		spec:                     spec,
+		backgroundProcessManager: backgroundProcessManager,
+		cpuStressArgsBuilder:     argsBuilder,
+	}
 }
 
 func (i *cpuPressureInjector) GetDisruptionKind() types.DisruptionKindName {
 	return types.DisruptionKindCPUPressure
 }
 
 func (i *cpuPressureInjector) Inject() error {
-	// move the injector process to the target cgroup controllers
-	i.config.Log.Infow("joining target cgroups")
-
-	if err := i.config.Cgroup.Join(i.config.ProcessManager.ProcessID()); err != nil {
-		return fmt.Errorf("failed to join the target cgroup: %w", err)
-	}
+	i.config.Log.Infow("creating processes to stress target", "count", i.spec.Count)
 
-	// get cores to stress from cpuset config
-	cores, err := i.config.StresserManager.TrackInjectorCores(i.config.Cgroup, i.spec.Count)
-	if err != nil {
-		return fmt.Errorf("failed to parse CPUSet %w", err)
-	}
+	var (
+		percentage int
+		err        error
+	)
 
-	wg := sync.WaitGroup{}
-	succeeded := true
-	mutex := sync.Mutex{}
-
-	// create one stress goroutine per allocated core
-	// each goroutine is locked on its current thread, without any other routines running on it
-	// it allows to have a 1 routine = 1 thread pattern
-	// each thread is then moved to the target cpu and cpuset cgroups so it can be schedule on the target allocated cores
-	// each thread is also niced to the highest priority
-	// because of linux scheduling, each thread will occupy a different core of allocated cores when stressing the cpu
-	for _, core := range cores.ToSlice() {
-		if i.config.StresserManager.IsCoreAlreadyStressed(core) {
-			i.config.Log.Infof("core %d is already stressed, skipping", core)
-			continue
+	if i.spec.Count != nil && i.spec.Count.Type == intstr.Int { // if a number is provided, calculate a percentage against the amount of cpus assigned to current target
+		assignedCPUs, err := i.config.Cgroup.ReadCPUSet()
+		if err != nil {
+			return fmt.Errorf("unable to read CPUSet for current container: %w", err)
 		}
 
-		wg.Add(1)
-
-		go func(core int) {
-			// lock the routine on the current thread
-			runtime.LockOSThread()
-			defer runtime.UnlockOSThread()
-
-			// retrieve current thread PID
-			tid := i.config.ProcessManager.ThreadID()
-
-			if i.prioritizeStresserProcess(core, tid) {
-				mutex.Lock()
-				i.routines++
-				mutex.Unlock()
-				wg.Done()
-				i.startStresser(core, tid)
-			} else {
-				succeeded = false
-				wg.Done()
-			}
-		}(core)
+		percentage = int(math.Floor(float64(i.spec.Count.IntValue()) / float64(assignedCPUs.Size()) * 100))
+
+		i.config.Log.Infow("percentage calculated from number of cpu", "provided_value", i.spec.Count, "assigned_cpus", assignedCPUs, "percentage", percentage)
+	} else if percentage, err = intstr.GetScaledValueFromIntOrPercent(i.spec.Count, 100, true); err != nil { // if a percentage is provided, keep it as is
+		return fmt.Errorf("unable to caculate stress percentage for '%s': %w", i.spec.Count, err)
```

Comment:
```suggestion
		return fmt.Errorf("unable to calculate stress percentage for '%s': %w", i.spec.Count, err)
```

Noticed this typo in the tests